zetaclient/bitcoin/bitcoin_signer_test.go

```diff
@@ -451,6 +245,139 @@ func createTestClient(t *testing.T) *BTCChainClient {
 	return client
 }
 
+func TestAddWithdrawTxOutputs(t *testing.T) {
+	// Create test signer and receiver address
+	signer, err := NewBTCSigner(config.BTCConfig{}, stub.NewTSSMainnet(), clientcommon.DefaultLoggers(), &metrics.TelemetryServer{}, nil)
+	require.NoError(t, err)
+
+	// tss address and script
+	tssAddr := signer.tssSigner.BTCAddressWitnessPubkeyHash()
+	tssScript, err := PayToAddrScript(tssAddr)
+	require.NoError(t, err)
+	fmt.Printf("tss address: %s", tssAddr.EncodeAddress())
+
+	// receiver addresses
+	receiver := "bc1qaxf82vyzy8y80v000e7t64gpten7gawewzu42y"
+	to, err := common.DecodeBtcAddress(receiver, common.BtcMainnetChain().ChainId)
+	require.NoError(t, err)
+	toScript, err := PayToAddrScript(to)
+	require.NoError(t, err)
+
+	// test cases
+	tests := []struct {
+		name     string
+		tx       *wire.MsgTx
+		to       btcutil.Address
+		total    float64
+		amount   float64
+		nonce    int64
+		fees     *big.Int
+		cancelTx bool
+		fail     bool
+		message  string
+		txout    []*wire.TxOut
+	}{
+		{
```

Comment:
Seems to me we could also have a case `len(txout) == 1` ?
If cancelTx == true and remaining == 0 ?

Reply:
`len(txout) == 1` is an impossible case because the amount will be added to the remaining (change to TSS) when the tx gets canceled. I can add a `len(txout) == 2` with `cancelTx == false && remainingSats == 0`

Reply:
https://github.com/zeta-chain/node/pull/1942/commits/2bcdfbc727c12612159ffbf008bb00505a996b4b